db-functions

```diff
@@ -377,6 +377,27 @@ check_pkgrepos() {
 	return 0
 }
 
+check_stagingrepos() {
+	local pkgfile=${1}
+	local pkgrepo=${2}
+	local pkgname=$(getpkgname "${pkgfile}")
+	local pkgarch=$(getpkgarch "${pkgfile}")
+	local candidate
+
+	if in_array "${pkgrepo}" "${STABLE_REPOS[@]}"; then
+		candidate=$(find_repo_for_package "${pkgname}" "${pkgarch}" "${TESTING_REPOS[@]}") || return 0
```

Comment:
This seems like it should also look in $STAGING_REPOS. What if a user tries to add a package to a stable repo and a version of that package also exists in a staging repo? Shouldn't that also throw an  error?

Reply:
I forgot this, pushed some new commits to attest to the problem and fix it.